### PR TITLE
Upload download

### PR DESCRIPTION
**Pull request by eitansuez**
this should address issue https://github.com/pivotal-cf/azure-blobstore-resource/issues/19

### CODE REVIEW

File: azure/client.go [hunk 1]
```diff
@@ -108,85 +111,43 @@ func (c Client) Get(blobName string, snapshot time.Time) ([]byte, error) {
...
-		return nil, err
-	}
+// DownloadToFile download specified blobName to specified file
+func (c Client) DownloadBlobToFile(blobName string, file *os.File) error {
 
-	var snapshotPtr *time.Time
-	if !snapshot.IsZero() {
-		snapshotPtr = &snapshot
-	}
+	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s/%s", c.storageAccountName, c.container, blobName))
```
Comment (christianang) [inline azure/client.go]: Unfortunately, you can't assume `blob.core.windows.net`. This needs to be changed to reference the `c.baseURL` instead. Also preferably we wouldn't throw the error away and return the error if it is malformed.

File: azure/client.go [hunk 2]
```diff
@@ -108,85 +111,43 @@ func (c Client) Get(blobName string, snapshot time.Time) ([]byte, error) {
...
 func (c Client) UploadFromStream(blobName string, stream io.Reader) error {
-	client, err := storage.NewClient(c.storageAccountName, c.storageAccountKey, c.baseURL, storage.DefaultAPIVersion, true)
-	if err != nil {
-		return err
-	}
 
-	blobClient := client.GetBlobService()
-	cnt := blobClient.GetContainerReference(c.container)
-	blob := cnt.GetBlobReference(blobName)
+	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s/%s", c.storageAccountName, c.container, blobName))
```
Comment (christianang) [inline azure/client.go]: Same as above. Need to use `baseURL` and return the error if it occurs.

**Comment by eitansuez** [comment]
hi christian. thanks for the review and the feedback.  will do.  i'll let you know when i have those items fixed and ready for you to merge.

**Comment by christianang** [comment]
Thanks for making those changes. Merged manually to rebase onto master.

**Pull request closed without merging**